src/python/supply/supply.go

```diff
@@ -494,24 +479,33 @@ func pipfileToRequirements(lockFilePath string) (string, error) {
 }
 
 func (s *Supplier) HandlePylibmc() error {
-	memcachedDir := filepath.Join(s.Stager.DepDir(), "libmemcache")
+	exists, err := s.Requirements.FindAnyPackage(s.Stager.BuildDir(), "pylibmc")
+	if err != nil {
+		return err
+	}
 
-	if err := s.Command.Execute(s.Stager.BuildDir(), ioutil.Discard, ioutil.Discard, "pip-grep", "-s", "requirements.txt", "pylibmc"); err == nil {
-		s.Log.BeginStep("Noticed pylibmc. Bootstrapping libmemcached.")
-		if err := s.Installer.InstallOnlyVersion("libmemcache", memcachedDir); err != nil {
-			return err
-		}
-		os.Setenv("LIBMEMCACHED", memcachedDir)
-		s.Stager.WriteEnvFile("LIBMEMCACHED", memcachedDir)
-		s.Stager.LinkDirectoryInDepDir(filepath.Join(memcachedDir, "lib"), "lib")
-		s.Stager.LinkDirectoryInDepDir(filepath.Join(memcachedDir, "lib", "sasl2"), "lib")
-		s.Stager.LinkDirectoryInDepDir(filepath.Join(memcachedDir, "lib", "pkgconfig"), "pkgconfig")
-		s.Stager.LinkDirectoryInDepDir(filepath.Join(memcachedDir, "include"), "include")
+	if exists {
+		return s.installLibmemcache()
 	}
 
 	return nil
 }
 
+func (s *Supplier) installLibmemcache() error {
```

Comment:
You might be able to remove this extra function by updating `HandlePylibmc` to check for `!exists`, then simply unindent the main code:

```go
...
if !exists {
    return nil
}

memcachedDir := filepath.Join(s.Stager.DepDir(), "libmemcache")
...
```